gobblin-runtime/src/main/java/gobblin/runtime/JobState.java

```diff
@@ -471,4 +502,62 @@ public JobExecutionInfo toJobExecutionInfo() {
 
     return jobExecutionInfo;
   }
+
+  /**
+   * Create a new {@link JobState.DatasetState} based on this {@link JobState} instance.
+   *
+   * @return a new {@link JobState.DatasetState} object
+   */
+  public DatasetState newDatasetState(boolean addTaskStates) {
+    DatasetState datasetState = new DatasetState(this.jobName, this.jobId);
+    datasetState.addAll(this);
+    datasetState.setState(this.state);
+    if (addTaskStates) {
+      datasetState.addTaskStates(this.taskStates.values());
+    }
+    return datasetState;
+  }
+
+  private static List<WorkUnitState> workUnitStatesFromDatasetStates(Iterable<JobState.DatasetState> datasetStates) {
+    ImmutableList.Builder<WorkUnitState> taskStateBuilder = ImmutableList.builder();
+    for (JobState datasetState : datasetStates) {
+      taskStateBuilder.addAll(datasetState.getTaskStatesAsWorkUnitStates());
+    }
+    return taskStateBuilder.build();
+  }
+
+  /**
+   * A subclass of {@link JobState} that is used to represent dataset states. This class is currently
+   * identical to {@link JobState} except that the name is more meaningful and less confusing.
+   */
+  public static class DatasetState extends JobState {
+
+    // For serialization/deserialization
+    @SuppressWarnings("unused")
+    public DatasetState() {
+      super();
+    }
+
+    public DatasetState(String jobName, String jobId) {
+      super(jobName, jobId);
+    }
+
+    public void setDatasetUrn(String datasetUrn) {
+      setProp(ConfigurationKeys.DATASET_URN_KEY, datasetUrn);
+    }
+
+    public String getDatasetUrn() {
+      return getProp(ConfigurationKeys.DATASET_URN_KEY, ConfigurationKeys.DEFAULT_DATASET_URN);
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
```

Comment:
Do you need to override these two methods since they just call the super methods?